lib/matplotlib/font_manager.py

```diff
@@ -855,7 +847,6 @@ def set_weight(self, weight):
         except ValueError:
             if weight not in weight_dict:
                 raise ValueError("weight is invalid")
-            weight = weight_dict[weight]
```

Comment:
I think this is the most concerning change in the PR.  Now the type of `self._weight` is either a string or an int, where as before it was consistently an int.